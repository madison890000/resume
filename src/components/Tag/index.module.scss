.filled {
  display: inline-block;
  margin: var(--base-gap-2-unit) var(--base-gap-2-unit);
  padding: var(--base-gap-2-unit) var(--base-gap-2-unit);
  background: gray;
  color: white;
  font-size: var(--base-font-size-middle);
  border-radius: var(--base-border-radius);

  @media screen and (max-width: 414px) {
    margin: var(--base-gap-unit) var(--base-gap-unit);
    padding: var(--base-gap-unit) var(--base-gap-unit);
  }
}

.normal {
  display: inline-block;
  margin: var(--base-gap-unit) var(--base-gap-2-unit);
  padding: var(--base-gap-unit) var(--base-gap-2-unit);
  border: 1px solid gray;
  color: gray;
  font-size: var(--base-font-size-small);
  border-radius: var(--base-border-radius);

  @media screen and (max-width: 414px) {
    margin: var(--base-gap-unit) var(--base-gap-unit);
    padding: var(--base-gap-unit) var(--base-gap-unit);
  }
}

.less {
  display: inline-block;
  margin: var(--base-gap-unit) var(--base-gap-unit);
  padding: var(--base-gap-unit) var(--base-gap-unit);
  color: gray;
  font-size: var(--base-font-size-small);
  border-radius: var(--base-border-radius);
  font-style: italic;

  @media screen and (max-width: 414px) {
    margin: var(--base-gap-unit) var(--base-gap-unit);
    padding: var(--base-gap-unit) var(--base-gap-unit);
  }
}
