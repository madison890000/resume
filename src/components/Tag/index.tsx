import React, { PropsWithChildren } from 'react';
import styles from './index.module.scss'

type TagType = 'filled' | 'normal' | 'less';

interface TagProps {
    type?: TagType
}

const Tag: React.FC<PropsWithChildren<TagProps>> = ({ type = 'normal', children }) => {
    switch (type) {
        case 'filled':
            return (
                <span className={styles.filled}>{children}</span>
            );
        case 'normal':
            return (
                <span className={styles.normal}>{children}</span>
            );
        case 'less':
            return (
                <span className={styles.less}>{children}</span>
            );
        default:
            return (
                <span>{children}</span>
            )
    }
}

export default Tag
