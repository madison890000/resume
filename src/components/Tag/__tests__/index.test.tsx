import React from 'react';
import Tag from '../index';
import renderer from 'react-test-renderer';

test('renders Tag', () => {
    const TagDom = renderer.create(
        <Tag>React</Tag>
    )
        .toJSON();
    expect(TagDom)
        .toMatchSnapshot();
});
test('renders Tag filled', () => {
    const TagDom = renderer.create(
        <Tag type="filled">React</Tag>
    )
        .toJSON();
    expect(TagDom)
        .toMatchSnapshot();
});
test('renders Tag less', () => {
    const TagDom = renderer.create(
        <Tag type="less">React</Tag>
    )
        .toJSON();
    expect(TagDom)
        .toMatchSnapshot();
});
