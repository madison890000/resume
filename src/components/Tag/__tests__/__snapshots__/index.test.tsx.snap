// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tag 1`] = `
<span
  className="normal"
>
  React
</span>
`;

exports[`renders Tag filled 1`] = `
<span
  className="filled"
>
  React
</span>
`;

exports[`renders Tag less 1`] = `
<span
  className="less"
>
  React
</span>
`;
